### PR TITLE
Enable ruff pydocstyle rules

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Enables ruff [pydocstyle](https://docs.astral.sh/ruff/rules/#pydocstyle-d), resolves violations of it, and disables rules _requiring_ docstrings (way too many violations of this).

Eventually we should run ruff on _changed_ files only and remove the ignored rules.

### CODE REVIEW

File: griptape/drivers/embedding/voyageai_embedding_driver.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 @define
 class VoyageAiEmbeddingDriver(BaseEmbeddingDriver):
-    """
+    """VoyageAI Embedding Driver.
```
Comment (dylanholmes) [inline griptape/drivers/embedding/voyageai_embedding_driver.py:11]: Not quite consistent with the other docstrings. For example the one for `AmazonS3FileManagerDriver` starts with a short sentence describing the classes purpose and does not split the class name.

Just pointing it out, but I'll approve anyway.

File: pyproject.toml [hunk 2]
```diff
@@ -209,14 +209,25 @@ select = [
...
     "B009", # get-attr-with-constant
     "B010", # set-attr-with-constant
+    "D100", # undocumented-public-module
+    "D101", # undocumented-public-class
+    "D102", # undocumented-public-method
+    "D103", # undocumented-public-function
+    "D104", # undocumented-public-package
+    "D105", # undocumented-magic-method
+    "D106", # undocumented-public-nested-class
+    "D107", # undocumented-public-init
```
Comment (dylanholmes) [inline pyproject.toml:227]: Out of curiosity were you planning on un-ignoring any of these later/incrementally?

**Review by dylanholmes — APPROVED** [review]

File: griptape/drivers/embedding/voyageai_embedding_driver.py (hunk 1 above)
Comment (collindutter) [reply griptape/drivers/embedding/voyageai_embedding_driver.py:11]: Yeah it's not ideal, I just wanted to add _something_ to conform to the linter.

File: pyproject.toml (hunk 2 above)
Comment (collindutter) [reply pyproject.toml:228]: Yeah, eventually we should run ruff on changed files only and remove the ignored rules.

**Review by dylanholmes — APPROVED** [review]

**Pull request merged**